pontoon/sync/core/translations_from_repo.py

```diff
@@ -433,10 +435,31 @@ def update_db_translations(
             f"[{project.slug}] Created {str_n_translations(created)} from repo changes"
         )
 
+    badge_update = ("", 0)
     if actions:
+        translation_before_level = log_user.badges_translation_level
+        review_before_level = log_user.badges_review_level
+
         ActionLog.objects.bulk_create(actions)
 
-    return created, list(suggestions.values())
+        if (
+            log_user.badges_translation_level > translation_before_level
+            and log_user.username != "pontoon-sync"
```

Comment:
You could add `if log_user.username == "pontoon-sync"` above this if, so you don't need to write the condition twice.